markdown_lab/network/throttle.py

```
@@ -0,0 +1,425 @@
+"""
+Advanced rate limiting with token bucket algorithm.
+
+This module provides sophisticated rate limiting with burst support,
+per-domain limits, and async compatibility for high-performance 
+request throttling.
+"""
+
+import asyncio
+import time
+import logging
+from typing import Dict, Optional, Union
```

Ruff (0.11.9)

`typing.Union` imported but unused

Remove unused import: `typing.Union`

(F401)